tests/js/server/password-function-tests.js

```diff
@@ -0,0 +1,84 @@
+/*
+
+  Tests for the password encoding static functions.
+
+ */
+"use strict";
+var fluid  = require("infusion");
+var gpii   = fluid.registerNamespace("gpii");
+var jqUnit = require("node-jqunit");
+
+require("../../../src/js/server/lib/password");
+
+fluid.registerNamespace("gpii.express.user.password.tests");
+gpii.express.user.password.tests.runTests = function (that) {
+    jqUnit.module("Unit test password encoding functions...");
+
+    jqUnit.test("Confirm that a range of known passwords are encoded as expected with full arguments...", function () {
+        fluid.each(that.options.expected, function (options, password) {
+            var actualEncodedString = gpii.express.user.password.encode(password, options.salt, options.iterations, 20);
+            jqUnit.assertEquals("The password should have been encoded correctly...", options.derived_key, actualEncodedString);
+        });
+    });
+
+    jqUnit.test("Confirm that a range of known passwords are encoded as expected with the defaults...", function () {
+        fluid.each(that.options.expected, function (options, password) {
+            var actualEncodedString = gpii.express.user.password.encode(password, options.salt);
+            jqUnit.assertEquals("The password should have been encoded correctly...", options.derived_key, actualEncodedString);
+        });
+    });
+
+    jqUnit.test("Confirm that a known password with other arguments is handled correctly...", function () {
+        fluid.each(that.options.oddball, function (options, password) {
+            var actualEncodedString = gpii.express.user.password.encode(password, options.salt, options.iterations, options.keyLength, options.digest);
+            jqUnit.assertEquals("The password should have been encoded correctly...", options.derived_key, actualEncodedString);
+        });
+    });
+
+    jqUnit.test("Confirm that hashes are generated as expected...", function () {
+        var saltString = gpii.express.user.password.generateSalt(24);
+        jqUnit.assertEquals("The salt string should be of twice the byte length...", 48, saltString.length);
+
+        var saltInt = parseInt(saltString, 16);
+        jqUnit.assertNotEquals("The salt string should evaluate to a number when parsed as hex...", NaN, saltInt);
```

Comment:
I'm not convinced this test tests anything since every number compares unequal to NaN. You can try https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger which is supported on node since before 0.12